lib/DDG/Spice/SalesTax.pm

```diff
@@ -4,41 +4,46 @@ package DDG::Spice::SalesTax;
 use strict;
 use DDG::Spice;
 use Locale::SubCountry;
-use YAML::XS 'LoadFile';
 
-triggers any => 'sales tax for', 'sales tax', 'sales tax in';
-spice from => '(.*)/(.*)';
-spice to => 'https://taxrates.api.avalara.com/postal?country=usa&postal=$1&apikey={{ENV{DDG_SPICE_SALESTAX_APIKEY}}}';
+triggers any => 'sales tax for', 'sales tax in', 'sales tax';
+spice to => 'http://dev.snapcx.io:9100/taxv1/getStateTaxRate?request_id=DDG_Requestor&state=$1';
 spice wrap_jsonp_callback => 1;
+spice is_cached => 1;
 
 #Create US SubCountry object
 my $US = new Locale::SubCountry("US");
 
-#State to Zip Code
-my $zipcodes = LoadFile(share('zipcodes.yml'));
-
 # Handle statement
 handle remainder_lc => sub {
-    my ($state,$zip); #Define vars
+    my ($state, $stateCode); #Define vars
     s/^what is (the)?//g; # strip common words
-    return unless $_; # Guard against "no answer"
+    return '' unless $_; # Guard against "no answer"
+    if ($_ =~ /^[0-9]{5}$/){
+      return;
+    }
+
+    if ($_ =~ /((?i)holiday)/){
+      return;
+    }
+
 
     # Washington D.C is a district and is not supported by the SubCountry package.
     if(m/\b(washington\s(dc|d\.c))\b/i) {
         $state = "Washington D.C";
+        $stateCode = "DC";
     } else {
         # $US->full_name returns the full state name based on the ISO3166 code
+        $stateCode = $_;
         $state = $US->full_name($_); # Check for state using ISO code (PA)
         if($state eq "unknown") {
             $state = $US->full_name($US->code($_)); # If state is "unknown" search for code using full state name (Pennsylvania)
+            $stateCode = $US->code($_);
         }
     }
 
-    $zip = $zipcodes->{$state}; # state name=> zip code
-
     # error checking
-    return if $state eq "unknown";
-    return unless (defined $state and defined $zip);
-    return $zip, $state; # return result
+    return '' if $state eq "unknown";
+    return '' unless (defined $state and $stateCode);
```

Comment:
We should only return the list of US State taxes for queries that come from the US. Anyone outside the US would find those results irrelevant.

We can perform a check like so: `if $loc && $loc->country_code eq 'US'`